Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,81 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            DateTime PostTime, CurrentTime, min = DateTime.Parse("00:00:00"), max = DateTime.Parse("23:59:59");
+            string ShowTime;
+
+            for(int i=0; i<exactPostTime.Length-1; i++)
+            {
+                for(int j=i+1; j<exactPostTime.Length; j++)
+                {
+                    if(exactPostTime[i] == exactPostTime[j])
+                    {
+                        if(showPostTime[i] != showPostTime[j])
+                        {
+                            return "impossible";
+                        }
+                    }
+                }
+            }
+
+            for(int i=0; i<exactPostTime.Length; i++)
+            {
+                PostTime = DateTime.Parse(exactPostTime[i]);
+                CurrentTime = PostTime;
+                ShowTime = showPostTime[i];
+                string[] split_ShowTime = ShowTime.Split(' ');
+
+                if (split_ShowTime[1].Equals("seconds"))
+                {
+                    CurrentTime = PostTime;
+                    if (DateTime.Compare(DateTime.Parse(CurrentTime.ToLongTimeString()), DateTime.Parse(min.ToLongTimeString())) == 1)
+                    {

```

Comment:
Also actions to be taken in case of hours, minutes or seconds are different and they could be moved to separate method.
Always, try to make your method/function as clean and small as possible.
That, doesn't mean you should always try to break a method into pieces even if separation is not possible.
But, you can always try to make it look cleaner